lib/numtheor.gi

```diff
@@ -1384,6 +1384,19 @@ InstallGlobalFunction( TwoSquares, function ( n )
 end );
 
 
+InstallGlobalFunction(PPart,function(n,p)
+local a;
+  if not IsPrimeInt(p) or not IsPosInt(n) then
+    Error("wrong parameters");
+  fi;
+  a:=1;
+  while n mod p=0 do
+    a:=a*p;
+    n:=n/p;
+  od;
+  return a;
+end);
+
```

Comment:
Hm, what for is `PPart` used? I also note that this is neither a performance exnhancements nor a cleanup. So I am not sure why it's in this PR? At the very least, some test cases would be nice.

If this is for a new package, then shouldn't it be in that package (I also note that while this function has a documentation comment, it is not hooked up to the manual, so to users, it will appear as undocuemnted)

That said, I do think a function in roughly this vein would be interesting for various people, me included -- e.g. in polycyclic, there is a related function `ValuationInt` (with a very inefficient implemention). The difference is that it does not return $p^k$, but just $k$, which strikes me as more flexible (just computing k, and then p^k, should be as efficient as this "direct" computation of p^k done here;  but computing k from p^k requires a log, and is relatively expensive).